editor/block-settings-menu/index.js

```diff
@@ -28,15 +28,19 @@ class BlockSettingsMenu extends Component {
 	}
 
 	toggleMenu() {
-		this.props.onSelect();
+		// Block could be hovered, not selected.
+		if ( this.props.uids.length === 1 ) {
+			this.props.onSelect( this.props.uids[ 0 ] );
```

Comment:
Not related to this PR but I guess this means we could drop this line https://github.com/WordPress/gutenberg/pull/2934/files#diff-1d765b1a62f4ddc8c86819016031b8dfR23

Reply:
Yeah, was wondering why there were two calls. Not sure which to delete. The call here is not strictly necessary, it's only needed for the inspector I guess.

Reply:
No, actually it's necessary when we click on the toggle for a "hovered" block, we need to select it first to avoid issues when trying to click the menu icons.

Reply:
This is done.